tiledb/sm/cpp_api/ndrectangle.h

```diff
@@ -47,6 +47,15 @@
 namespace tiledb {
 
 class NDRectangle {
+  template <class K, class Y = void>
+  struct tuple_ret {
+    using type = std::array<K, 2>;
+  };
+  template <class Y>
+  struct tuple_ret<std::string, Y> {
+    using type = std::array<std::string, 2>;
+  };
```

Comment:
This structure doesn't accomplish anything. When template argument `K` is `std::string` the resulting member alias `type` is always `std::array<std::string,2>` regardless of whether the specialization is present or absent.